intergov/processors/multichannel_router/__init__.py

```diff
@@ -0,0 +1,255 @@
+"""
+* Get pending messages from the ApiOutboxRepo
+* Mark them as 'sending' in the local storage
+* Pick a channel
+* Send them to the channel
+* Change their status in the message_rx_api to "accepted" (?)
+* Marks them as 'accepted' in the local storage
+
+local storage - postgres for example (and for the demo)
+"""
+import random
+import time
+
+from intergov.conf import env_postgres_config, env
+from intergov.channels.shared_db_channel import SharedDbChannel
+from intergov.repos.api_outbox import ApiOutboxRepo
+from intergov.repos.api_outbox.postgres_objects import Message as PostgresMessageRepr
+from intergov.loggers import logging
+from intergov.domain.wire_protocols import generic_discrete as gd
+from intergov.use_cases.route_to_channel import RouteToChannelUseCase
+
+logger = logging.getLogger('multichannel_router')
+
+# Commented out because we decided not to use memory channel for some time
+# # this is a kludge
+# # we need some kind of configured registry
+# DEFAULT_CONFIG = [
+#     {
+#         'id': 'DiscreteGenericMemoryChannel',
+#         'type': DiscreteGenericMemoryChannel,
+#     }
+# ]
+
+
+class MultichannelWorker(object):
+    """
+    Iterate over the RouteToChannelUseCase.
+    """
+
+    ROUTING_TABLE = [
+        {
+            "Name": "shared db channel to Australia",  # any text
+            # these 2 are for routing the messages
+            "Jurisdiction": "AU",  # please add more countries if you need to
+            "Predicate": "UN.CEFACT.",  # quite wide
+            # we want to make it look universal but don't spend
+            # too much time designing things we never need
+            # so ChannelType is just for initiating the correct channel helper class
+            # (which will be doing the real message send work)
+            "ChannelType": "SharedDbChannel",  # see intergov.channels.shared_db_channel for details
```

Comment:
See comment on the SharedDbChannel file about all channels implementing the same API and not needing to know the type.